File: src/main/java/cccs/oldversion/tag/TransitoryTag.java
package cccs.oldversion.tag;

import org.apache.spark.sql.Row;

public class TransitoryTag extends Tag {

    public static TransitoryTag TRUE = new TransitoryTag(true);
    public static TransitoryTag FALSE = new TransitoryTag(false);
    private final boolean value;

    private TransitoryTag(boolean value) {
        this.value = value;
    }

    public static TransitoryTag of(boolean value) {
        if (value) {
            return TRUE;
        }
        return FALSE;
    }

    @Override
    public boolean currentValue() {
        return value;
    }

    @Override
    public boolean evaluateRow(Row row) {
        log.debug("evaluating ImmutableTag");
        return currentValue();
    }

    @Override
    public void setCurrentValue(boolean value) {        
    }
}

File: src/main/scala/cccs/fluxcapacitor/tag/TransitoryTag.scala
package cccs.fluxcapacitor.tag

import org.apache.spark.sql.Row

class TransitoryTag() extends Tag {

  var value = false

  override def currentValue(): Boolean = value

  override def evaluateRow(row: Row) = {
    if (log.isTraceEnabled) log.trace("evaluating ImmutableTag")
    currentValue()
  }

  override def setCurrentValue(value: Boolean): Tag = {
    this.value = value
    this
  }
}
